internal/consumer/sqsConsumer.go

```diff
@@ -0,0 +1,182 @@
+/**
+* Copyright 2023-present Coinbase Global, Inc.
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*  http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+ */
+
+package consumer
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"strings"
+
+	"github.com/amzn/ion-go/ion"
+	"github.com/aws/aws-lambda-go/events"
+	"github.com/aws/aws-sdk-go-v2/service/kinesis/types"
+	"github.com/awslabs/kinesis-aggregation/go/v2/deaggregator"
+	"github.com/coinbase-samples/ib-ledger-go/internal/model"
+	"github.com/coinbase-samples/ib-ledger-go/internal/relationaldb"
+
+	log "github.com/sirupsen/logrus"
+)
+
+func Listen(l *log.Entry) {
+	ctx := context.Background()
+	for {
+
+		message, err := Repo.ReceiveSqsMsg(ctx, 90)
+
+		if err != nil {
+			l.Error(err)
+			continue
+		}
+
+		if message == nil {
+			continue
+		}
+
+		if err := onMessage(l, ctx, message.Body); err != nil {
+			l.Error(err)
+			continue
+		}
+
+		if err := Repo.DeleteSqsMsg(ctx, *message.ReceiptHandle); err != nil {
+			l.Error(err)
+			continue
+		}
+
+	}
+}
+
+func onMessage(l *log.Entry, ctx context.Context, body *string) error {
+	var kinesisEvent events.KinesisRecord
+	if err := json.Unmarshal([]byte(*body), &kinesisEvent); err != nil {
+		return fmt.Errorf("unable to parse sqs into kinesis event: %w", err)
+	}
+
+	kr := &types.Record{
+		ApproximateArrivalTimestamp: &kinesisEvent.ApproximateArrivalTimestamp.Time,
+		Data:                        kinesisEvent.Data,
+		EncryptionType:              getKinesisEncryptionType(kinesisEvent),
+		PartitionKey:                &kinesisEvent.PartitionKey,
+		SequenceNumber:              &kinesisEvent.SequenceNumber,
+	}
+
+	kcr := make([]types.Record, 0, 1)
+
+	kcr = append(kcr, *kr)
+
+	dars, _ := deaggregator.DeaggregateRecords(kcr)
+	var data map[string]interface{}
+
+	for _, r := range dars {
```

Comment:
Need to break this up into functions. For loop + if/else + switch statements all in the same block

Reply:
Yeah definitely. I'll split it all up.